betterjails/src/main/java/io/github/emilyydev/betterjails/commands/CommandHandler.java

```diff
@@ -26,383 +26,317 @@
-      jailListMessageThingu.addAll(messages.jailListEntryFormatter().formatJailList(jails.values()));
+      buffer.add(messages.listJailsFunnyMessage());
+      buffer.addAll(messages.jailListEntryFormatter().formatJailList(jails.values()));
     }
 
-    sender.sendMessage(jailListMessageThingu.toArray(DUMMY_STRING_ARRAY));
+    sender.sendMessage(String.join("\n", buffer));
   }
 
-  private void releasePrisoner(final CommandSender sender, final String prisoner) {
-    final String executioner = sender.getName();
-    final OfflinePlayer player = this.server.getOfflinePlayer(
-        this.namesToUuid.getOrDefault(prisoner.toLowerCase(Locale.ROOT), Util.NIL_UUID)
-    );
-
-    if (player.getUniqueId().equals(Util.NIL_UUID)) {
-      sender.sendMessage(this.configuration.messages().releasePrisonerFailedNeverJoined(prisoner, executioner));
-      return;
-    }
-
-    final boolean wasReleased = this.plugin.prisonerData().releaseJailedPlayer(player, uuidOrNil(sender), executioner, true);
-    if (wasReleased) {
-      this.server.broadcast(
-          this.configuration.messages().releasePrisonerSuccess(prisoner, executioner),
-          "betterjails.receivebroadcast"
+  @Permission("betterjails.unjail")
+  @Command("unjail|release <prisoner>")
+  @CommandDescription("Releases an imprisoned player")
+  public void releasePrisoner(
+      final CommandContext<CommandSender> ctx,
+      final CommandSender sender,
+      final ApiPrisoner prisoner
+  ) {
+    final String executorName = sender.getName();
+    if (prisoner.released()) {
+      throw new CommandError(
+          ctx, CommandError.UNJAIL_FAILED_PLAYER_NOT_JAILED,
+          CommandError.prisonerVariable(prisoner.nameOr("(unknown)")),
+          CommandError.executorVariable(sender.getName())
       );
-    } else {
-      sender.sendMessage(this.configuration.messages().releasePrisonerFailedNotJailed(prisoner, executioner));
     }
-  }
 
-  private void createOrRenameJail(final CommandSender sender, final String jail) {
-    if (!(sender instanceof Player)) {
-      sender.sendMessage(this.configuration.messages().createJailFromConsole(sender.getName(), jail));
-      return;
-    }
+    this.plugin.prisonerData().releasePrisoner(prisoner, this.server.getOfflinePlayer(prisoner.uuid()), uuidOrNil(sender), executorName, true);
+    this.server.broadcast(
+        this.configuration.messages().releasePrisonerSuccess(prisoner.nameOr("(unknown)"), executorName),
+        "betterjails.receivebroadcast"
+    );
+  }
 
-    final Player player = (Player) sender;
-    this.plugin.jailData().addJail(jail, player.getLocation()).whenCompleteAsync((v, ex) -> {
+  @Permission("betterjails.setjail")
+  @Command(value = "setjail <name>", requiredSender = Player.class)
+  @CommandDescription("Creates a new jail or relocates an existing jail to where the command is executed")
+  public CompletableFuture<Void> createOrRelocateJail(final Player sender, final String name) {
```

Comment:
This command isn't even registered. No clue why.
![image](https://github.com/user-attachments/assets/32d67ad2-beaf-4406-b0cc-c1898ac7f770)